address/records.go

```diff
@@ -91,3 +94,16 @@ func newAddressAmountRecord(amount *uint64) tlv.Record {
 		asset.VarIntEncoder, asset.VarIntDecoder,
 	)
 }
+
+func newProofCourierAddrRecord(addr *string) tlv.Record {
+	addrBytes := []byte(*addr)
```

Comment:
I don't think this will work for decoding, as the result isn't written back into the `addr` pointer. Maybe we should instead make the addr just a `[]byte`?